src/Nethermind/Nethermind.Core/Encoding/TransactionReceiptDecoder.cs

```diff
@@ -101,12 +102,90 @@ public Rlp Encode(TxReceipt item, RlpBehaviors rlpBehaviors = RlpBehaviors.None)
 
         public void Encode(MemoryStream stream, TxReceipt item, RlpBehaviors rlpBehaviors = RlpBehaviors.None)
         {
-            throw new System.NotImplementedException();
+            if (item == null)
+            {
+                stream.Write(Rlp.OfEmptySequence.Bytes);
+                return;
+            }
+            
+            Rlp.StartSequence(stream, GetLength(item, rlpBehaviors));
+            if (rlpBehaviors.HasFlag(RlpBehaviors.Eip658Receipts))
+            {
+                Rlp.Encode(stream, item.StatusCode);
+            }
+            else
+            {
+                Rlp.Encode(stream, item.PostTransactionState);
+            }
+
+            if (rlpBehaviors.HasFlag(RlpBehaviors.Storage))
+            {
+                Rlp.Encode(stream, item.BlockHash);
+                Rlp.Encode(stream, item.BlockNumber);
+                Rlp.Encode(stream, item.Index);
+                Rlp.Encode(stream, item.Sender);
+                Rlp.Encode(stream, item.Recipient);
+                Rlp.Encode(stream, item.ContractAddress);
+                Rlp.Encode(stream, item.GasUsed);
+                Rlp.Encode(stream, item.GasUsedTotal);
+                Rlp.Encode(stream, item.Bloom);
+                Rlp.Encode(stream, item.Logs);
+                Rlp.Encode(stream, item.Error);
+            }
+            else
+            {
+                Rlp.Encode(stream, item.GasUsedTotal);
+                Rlp.Encode(stream, item.Bloom);
+                Rlp.Encode(stream, item.Logs);
+            }
+        }
+        
+        private int GetContentLength(TxReceipt item, RlpBehaviors rlpBehaviors)
+        {
+            var contentLength = 0;
+            if (item == null)
+            {
+                return contentLength;
+            }
+            
+            if (rlpBehaviors.HasFlag(RlpBehaviors.Storage))
```

Comment:
HasFlag has very bad implementation internally and it has *terrible* performance. Please use & or I operators instead.

Reply:
https://stackoverflow.com/questions/7368652/what-is-it-that-makes-enum-hasflag-so-slow

Reply:
Usage of HasFlag was once a hot path on our performance benchmarks.